# Model Inference

---

Nama : Rajib Kurniawan

Batch : HCK-032

**Objective** : Membuat Model Machine Learning yang akan dibuat dengan menggunakan beberapa pemodelan dan dianalisa mana model yang paling bagus dan baik untuk dataset ini sehingga akan didapatkan result atau predict yang akurat.

---

# Import Libraries

In [1]:
# Import libraries

import pickle
import pandas as pd


# Load Model

In [2]:
# # Load model terbaik
model_path = "best_model_xgboost.pkl"
model = pickle.load(open(model_path, "rb"))

print("✅ Model berhasil dimuat kembali!")

✅ Model berhasil dimuat kembali!


# Inference

In [3]:
# Data new user (pastikan nama kolom SAMA dengan data training!)
new_user = {
    "Age": [28],
    "Gender": ["male"],
    "Education_Level": ["Bachelor"],
    "Income": [52000],
    "Experience_Years": [5],
    "Home_Status": ["RENT"],
    "Loan_Amount": [8000],
    "Loan_Purpose": ["EDUCATION"],
    "Loan_Interest_Rate": [11.2],
    "Loan_to_Income_Ratio": [0.15],
    "Credit_History_Length": [4],
    "Credit_Score": [710],
    "Default_On_Payment": ["No"],
}

# Buat DataFrame
data_inf = pd.DataFrame(new_user)

# Kalau di training kamu buat fitur turunan Experience_Category, tambahkan juga:
def categorize_experience(years):
    if years <= 2:
        return "Entry Level"
    elif years <= 5:
        return "Junior"
    elif years <= 10:
        return "Senior"
    elif years <= 20:
        return "Expert"
    else:
        return "Veteran"


data_inf["Experience_Category"] = data_inf["Experience_Years"].apply(
    categorize_experience
)

In [4]:
# Prediksi
pred = model.predict(data_inf)
proba = model.predict_proba(data_inf)[:, 1]

In [5]:
# Tampilkan hasil
data_inf["Loan_Status_Prediction"] = pred
data_inf["Loan_Approval_Probability"] = proba
data_inf

Age Gender Education_Level  Income  Experience_Years Home_Status  \
0   28   male        Bachelor   52000                 5        RENT   

   Loan_Amount Loan_Purpose  Loan_Interest_Rate  Loan_to_Income_Ratio  \
0         8000    EDUCATION                11.2                  0.15   

   Credit_History_Length  Credit_Score Default_On_Payment Experience_Category  \
0                      4           710                 No              Junior   

   Loan_Status_Prediction  Loan_Approval_Probability  
0                       1                   0.965934

Kesimpulan Inference :
Berdasarkan hasil inference di atas, model XGBoost terbaik yang telah dilatih pada data historis pengajuan pinjaman memberikan hasil sebagai berikut:
- Prediksi Status Pinjaman: 1 -> artinya pengajuan pinjaman disetujui.
- Probabilitas Persetujuan: 96.59% -> model memiliki tingkat keyakinan tinggi bahwa calon peminjam memenuhi kriteria kelayakan kredit.

Analisis singkat terhadap profil peminjam baru:
- Usia 28 tahun dengan tingkat pendidikan Bachelor dan penghasilan tahunan sebesar 52.000.
- Telah memiliki pengalaman kerja selama 5 tahun (Junior level).
- Status rumah masih RENT (menyewa), namun skor kredit cukup tinggi (710) dan rasio pinjaman terhadap pendapatan rendah (0.15).
- Tidak memiliki riwayat gagal bayar sebelumnya (Default_On_Payment = No).
Berdasarkan kombinasi faktor-faktor tersebut, model memprediksi bahwa peminjam ini memiliki risiko gagal bayar yang sangat rendah, sehingga layak untuk mendapatkan persetujuan pinjaman.

Insight yang didapatkan:
- Faktor-faktor dengan pengaruh besar terhadap keputusan approval umumnya adalah Credit Score, Income, Loan-to-Income Ratio, dan Default_On_Payment.
- Peminjam dengan skor kredit tinggi, pendapatan stabil, dan rasio pinjaman rendah memiliki kemungkinan besar disetujui.
- Fitur turunan seperti Experience_Category juga membantu memberikan konteks tambahan terhadap stabilitas pekerjaan peminjam.

Kesimpulan Final :
Jadi berdasarkan hasil inferensi, model XGBoost mampu memprediksi secara akurat dan logis, dengan hasil bahwa calon peminjam ini layak mendapatkan persetujuan pinjaman.